### PR TITLE
clear avatar cache on user change

### PR DESCRIPTION
**Pull request by chrisnojima**
steps to repro on master

1. log in
1. view user A
1. switch to a different user
1. view user A, see a grey avatar

@keybase/react-hackers cc: @mmaxim

### CODE REVIEW

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump: 10 line change

File: shared/common-adapters/avatar.js [hunk 1]
```diff
@@ -156,6 +165,7 @@ const _sharedAskForUserData = new SharedAskForUserData()
 
 const mapStateToProps = (state: TypedState, ownProps: OwnProps) => {
   const name = ownProps.username || ownProps.teamname
+  _sharedAskForUserData._checkLoggedIn(name)
```
Comment (mmaxim) [inline shared/common-adapters/avatar.js]: This is kind of confusing to me, we could be passing in a team name to check against the current username?

Comment (chrisnojima) [reply shared/common-adapters/avatar.js]: this is us checking if our own username has changed (aka we logged in / out)

Comment (chrisnojima) [reply shared/common-adapters/avatar.js]: oh this got mutated, needs to be state.config.username

Comment (mmaxim) [reply shared/common-adapters/avatar.js]: K that makes more sense, thanks!

Comment (chrisnojima) [reply shared/common-adapters/avatar.js]: updated tnx!

**Pull request merged**